session.go

```diff
@@ -995,17 +1054,22 @@ func (st *StmtTxn) cleanup() {
 }
 
 // StmtCommit implements the context.Context interface.
-func (s *session) StmtCommit() error {
-	defer s.StmtTxn.cleanup()
-	st := &s.StmtTxn
-	err := kv.WalkMemBuffer(s.buf, func(k kv.Key, v []byte) error {
+func (s *session) StmtCommit() {
+	if s.txn.fail != nil {
+		return
+	}
+
+	defer s.txn.cleanup()
+	st := &s.txn
+	err := kv.WalkMemBuffer(st.buf, func(k kv.Key, v []byte) error {
 		if len(v) == 0 {
 			return errors.Trace(st.Transaction.Delete(k))
 		}
 		return errors.Trace(st.Transaction.Set(k, v))
 	})
 	if err != nil {
-		return errors.Trace(err)
+		s.txn.fail = errors.Trace(err)
```

Comment:
Please add a test case for this.

Reply:
Done

Reply:
If err is occured, but some keys is already set to st.Transaction, (some error like ErrEntryTooLarge),  how to deal with these keys?

Reply:
The whole transaction will be rollback...

Reply:
but you just set `s.txn.fail = err`, but not return err, the whole transaction will be rollback?

Reply:
Yes, the description of this PR:

> If StmtCommit() meets error, mark it, rollback the whole transaction later.

But I bet `StmtCommit()` never meet error in practice.